doc/website/blog/2023-12-12-hibernate-schema-loader.mdx

```diff
@@ -0,0 +1,593 @@
+    }
+
+    @Id
+    public String name;
+
+    @OneToMany(mappedBy = "actor", cascade = CascadeType.PERSIST)
+    public Set<MovieParticipation> actedIn;
+}
+```
+
+</TabItem>
+<TabItem value="Movie.java">
+
+```java
+package org.example;
+
+import jakarta.persistence.*;
+
+import java.util.List;
+
+@Entity
+@Table(name = "Movies")
+public class Movie {
+    Movie(String title, Integer numberInSeries) {
+        this.title = title;
+        this.numberInSeries = numberInSeries;
+    }
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    public Long id;
+
+    public String title;
+
+    public Integer numberInSeries;
+}
+
+```
+
+</TabItem>
+<TabItem value="MovieParticipation.java">
+
+```java
+package org.example;
+
+import jakarta.persistence.*;
+
+@Entity
+public class MovieParticipation {
+    MovieParticipation(Movie movie, Actor actor) {
+        this.key = new MovieParticipationKey(movie.id, actor.name);
+        this.actor = actor;
+        this.movie = movie;
+    }
+
+    @EmbeddedId
+    public MovieParticipationKey key;
+
+    @ManyToOne(cascade=CascadeType.PERSIST)
+    @MapsId("movieId")
+    @JoinColumn(name = "movieId")
+    Movie movie;
+
+    @ManyToOne(cascade=CascadeType.PERSIST)
+    @MapsId("actorName")
+    @JoinColumn(name = "actorName")
+    Actor actor;
+}
+```
+
+</TabItem>
+<TabItem value="MovieParticipationKey.java">
+
+```java
+package org.example;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+
+import java.io.Serializable;
+
+@Embeddable
+public class MovieParticipationKey implements Serializable {
+    public MovieParticipationKey(Long movieId, String actorName) {
+        this.movieId = movieId;
+        this.actorName = actorName;
+    }
+    public Long movieId;
+
+    public String actorName;
+}
+```
+
+</TabItem>
+<TabItem value="Main.java">
+
+```java
+package org.example;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.MetadataSources;
+
+import java.util.Set;
+
+public class Main {
+    public static void main(String[] args) {
+        SessionFactory sessionFactory = new MetadataSources()
+                .addAnnotatedClass(Movie.class)
+                .addAnnotatedClass(Actor.class)
+                .addAnnotatedClass(MovieParticipation.class)
+                .buildMetadata()
+                .buildSessionFactory();
+
+        sessionFactory.inTransaction(session -> {
+                List<Actor> actors = session.createQuery("from Actor").list();
+                if (!actors.isEmpty()) {
+                actors.forEach(x -> System.out.println("Found Actor " + x.name));
+            } else {
+                Movie matrix = new Movie("The Matrix", 1);
+                Actor keanuReeves = new Actor("Keanu Reeves");
+                keanuReeves.actedIn = Set.of(new MovieParticipation(matrix, keanuReeves));
+                session.persist(keanuReeves);
+            }
+        });
+    }
+}
+```
+
+</TabItem>
+</Tabs>
+
```

Comment:
Start with visualize, then move to diff and add this note right after the command.